research-software-directory/RSD-as-a-service
# 1170 dedicated highlights page

# Dedicated highlights page

Fixes #1170

Changes proposed in this pull request:

* adds a dedicated page for software highlights
  ![image](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/9e631acc-5c0d-4f16-9251-4cf119a18719)
* the highlights are by default sorted "Latest featured", i.e. ascending position
* the filters in the sidebar show only properties of the highlights
* the software highlights page can be reached via a button below the highlights carousel
  ![image](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/1dbbc748-e135-4530-9d0b-a3d282da7bba)
* note that the configuration for software highlights in `settings.json` changes slightly

How to test:

* optionally: define `software_highlights` title and description in `settings.json`
* `docker compose down --volumes && docker compose build --parallel && docker compose up`
* in a second terminal window `docker compose run data-generation`
* navigate to software overview
* if there are no highlights, login as admin and add some
* click on the "Browse all Highlights" button below the highlights carousel
* verify that filters work correctly and offer options that correspond to the displayed highlights

Points to be regarded in followup PR:

* how to configure the amount of highlights shown in the carousel (e.g. settings.json vs switch in admnin frontend?)
* how to specify whether the "Browse all highlights" button is shown on software index page?
* as an administrator, does it make sense when adding a new highlight, the position will have the highest number and that it has to be moved manually to position 1?

PR Checklist:
